Behaviors/SoftDeleteBehavior.php

```diff
@@ -0,0 +1,211 @@
+                        $updateData->getColumns()->remove($col);
+                }
+            } else {
+                $updateData->getColumns()->remove($col);
+            }
+        }
+
+        // first check if metaobject has soft-delete-attribute
+        if ($eventData->getMetaObject()->hasAttribute($this->getSoftDeleteAttributeAlias())){
+            
+            // add the soft-delete-column
+            $deletedCol = $updateData->getColumns()->addFromAttribute($this->getSoftDeleteAttribute());
+            
+            // if there are no datarows in the passed datasheet, but there are filters assigned:
+            // add a single row of data, with only the soft-delete-attribute being set, so that this 
+            // attribute can be assigned to every row fitting the filter later
+            if ($updateData->isEmpty() && $updateData->getFilters() !== null){
+                $updateData->addRow([$this->getSoftDeleteAttributeAlias() => $this->getSoftDeleteValue()]);
+            }
+            
+            // if the datasheet still contains no datarows, then no items have to be marked as deleted
+            if ($updateData->isEmpty() === false){
+                $deletedCol->setValueOnAllRows($this->getSoftDeleteValue());
+                $updatedRows = $updateData->dataUpdate(false, $transaction);
+                $affected_rows += $updatedRows;
+            }
+        }
+            
+        $eventData->setCounterForRowsInDataSource($updateData->countRowsInDataSource());
+
+        // also update the original data sheet for further use
+        if ($eventData->isEmpty() === false){
+            $deletedCol = $eventData->getColumns()->getByAttribute($this->getSoftDeleteAttribute());
+            $deletedCol->setValueOnAllRows($this->getSoftDeleteValue());
+        }
+
+        return $affected_rows;
+    }
+    
+    /**
+     * 
+     * @return string
+     */
+    public function getSoftDeleteAttributeAlias() 
+    {
+        if ($this->getObject()->hasAttribute($this->soft_delete_attribute_alias)){
+            return $this->soft_delete_attribute_alias;
+        } else {
+            throw new BehaviorConfigurationError($this->getObject(), 'Configuration error: no attribute ' . $this->getSoftDeleteAttributeAlias() . 'found in object ' . $this->getObject()->getAlias() . '.');
+        }
+    }
+    
+    /**
+     * Alias of the attribute, where the deletion flag is being set.
+     *
+     * @uxon-property soft_delete_attribute_alias
+     * @uxon-type metamodel:attribute
+     * @uxon-required true
+     *
+     * @param string $value
+     * @return \exface\Core\Behaviors\SoftDeleteBehavior
+     */
+    public function setSoftDeleteAttributeAlias(string $value) : SoftDeleteBehavior
+    {
+        $this->soft_delete_attribute_alias = $value;
+        //check whether the getter throws en exeption, to ensure the attributename is valid for this object
+        $this->getSoftDeleteAttributeAlias();
```

Comment:
Bitte explizit object->hasAttribute() aufrufen. So wird es nur Probleme geben, da niemand auf die Idee kommen wird Bei Anderungen am getter in den setter reinzuschauen.